src/app/personside/infotabs/utbetalinger/utbetaling/EnkelUtbetaling.tsx

```diff
@@ -63,125 +50,110 @@ const UtbetalingHeaderStyle = styled.div`
+    const sum = formaterNOK(ytelse.nettobeløp);
+    const periode = periodeStringFromYtelse(ytelse);
+    const forfallsInfo = utbetaling.forfallsdato && !utbetaling.utbetalingsdato ? `Forfallsdato: ${dato}` : '';
+
+    const ekspanderYtelse = (ekspander: boolean) => dispatch(setEkspanderYtelse(ytelse, ekspander));
+    const setYtelseIFokus = () => dispatch(setNyYtelseIFokus(ytelse));
+
+    useOnMount(() => {
+        if (props.valgt) {
+            utbetalingRef.current && utbetalingRef.current.focus();
         }
-    }
+    });
 
-    componentDidUpdate(prevProps: Props) {
-        const fikkFokus = this.props.erIFokus && !prevProps.erIFokus;
-        if (fikkFokus && this.utbetalingRef.current) {
-            this.utbetalingRef.current.focus();
+    const prevErIFokus = usePrevious(erIFokus);
+    useEffect(() => {
+        const fikkFokus = erIFokus && !prevErIFokus;
+        if (fikkFokus && utbetalingRef.current) {
+            utbetalingRef.current.focus();
         }
-    }
+    }, [prevErIFokus, erIFokus, utbetalingRef]);
 
-    toggleVisDetaljer() {
-        this.props.setEkspanderYtelse(!this.props.visDetaljer);
-        this.props.setYtelseIFokus();
-    }
+    const toggleVisDetaljer = () => {
+        ekspanderYtelse(!visDetaljer);
+        setYtelseIFokus();
+    };
 
-    handlePrint() {
-        loggEvent('EnkeltUtbetaling', 'Printer');
-        this.props.setEkspanderYtelse(true);
-        this.print && this.print();
-    }
+    const handlePrint = () => {
+        ekspanderYtelse(true);
+        print();
+        loggEvent('EnkeltUtbetaling', 'usePrinter');
+    };
 
-    handleClickOnUtbetaling(event: React.MouseEvent<HTMLElement>) {
-        const printKnappTrykket = eventTagetIsInsideRef(event, this.printerButtonRef);
+    const handleClickOnUtbetaling = (event: React.MouseEvent<HTMLElement>) => {
+        const printKnappTrykket = eventTagetIsInsideRef(event, printerButtonRef);
         if (!printKnappTrykket) {
-            this.toggleVisDetaljer();
+            toggleVisDetaljer();
         }
-    }
-
-    render() {
-        const utbetaling = this.props.utbetaling;
-
-        const ytelse = this.props.ytelse;
-
-        const dato = datoVerbose(getGjeldendeDatoForUtbetaling(utbetaling)).sammensatt;
-        const tittel = ytelse.type;
-        const sum = formaterNOK(ytelse.nettobeløp);
-        const periode = periodeStringFromYtelse(ytelse);
-        const forfallsInfo = utbetaling.forfallsdato && !utbetaling.utbetalingsdato ? `Forfallsdato: ${dato}` : '';
-
-        return (
-            <Printer getPrintTrigger={(trigger: () => void) => (this.print = trigger)}>
-                <UtbetalingStyle
-                    onClick={(event: React.MouseEvent<HTMLElement>) =>
-                        cancelIfHighlighting(() => this.handleClickOnUtbetaling(event))
-                    }
-                    ref={this.utbetalingRef}
-                    tabIndex={0}
-                    onFocus={this.props.setYtelseIFokus}
```

Comment:
@winsvold denne fjernet jeg for å få onClicken til å fungere. Har du forslag på hvordan vi kan fikse dette på en ryddig måte?